src/libraries/Microsoft.Extensions.Configuration.UserSecrets/tests/ConfigurationExtensionTest.cs

```diff
@@ -107,6 +107,19 @@ public void AddUserSecrets_DoesNotThrowsIfOptional()
             Assert.Empty(config.AsEnumerable());
         }
 
+        [Fact]
+        public void AddUserSecrets_DoesThrowsIfNotOptionalAndSecretDoesNotExist()
+        {
+            var secretId = Assembly.GetExecutingAssembly().GetName().Name;
+            var secretPath = PathHelper.GetSecretsPathFromSecretsId(secretId);
+            if (File.Exists(secretPath))
+            {
+                File.Delete(secretPath);
+            }
+
+            Assert.Throws<FileNotFoundException>(() => new ConfigurationBuilder().AddUserSecrets(Assembly.GetExecutingAssembly(), false).Build());
```

Comment:
nit: ideally you'd only want to keep the method that you're asserting behavior within the Assert.Throws block.